logfire/_internal/formatter.py

```diff
@@ -450,3 +451,29 @@ def warn_inspect_arguments(msg: str, stacklevel: int):
     ) + msg
     warnings.warn(msg, InspectArgumentsFailedWarning, stacklevel=stacklevel)
     logfire.log('warn', msg)
+
+
+class KnownFormattingError(Exception):
+    """An error raised when there's something wrong with a format string or the field values.
+
+    In other words this should correspond to errors that would be raised when using `str.format`,
+    and generally indicate a user error, most likely that they weren't trying to pass a template string at all.
+    """
+
+
+class FormattingFailedWarning(Warning):
```

Comment:
```suggestion
class FormattingFailedWarning(UserWarning):
```

I think if we view a `FormattingFailedWarning` as a result of a mistake in user code, it would be marginally better for it to inherit from `UserWarning`. If not, the current implementation is fine.

It seems to me in practice, 9 times out of 10 that a `KnownFormattingError` is raised, it will be due to a "mistake" in user code, which is why I would suggest using `UserWarning` as the base class here but if I'm missing something or you just disagree with that change we can keep it as is.